### DISCUSSION TITLE
How to publish two test result set

### DISCUSSION THREAD

**Discussion by soroshsabz** [cake-build/cake#3137] [Q&A]
ITNOA

I have one test project that run with NUnit3 Runner in cake that generate test result in test1.xml with NUnit format, and I have another test project that run with VSTest Runner in cake that generate test result in test2.xml with VSTest format.

My question is How to publish two test result into Azure DevOps with one command? or any other way exist to combine these files?

My command is like below

		if (testProjects.Any() && AzurePipelines.IsRunningOnAzurePipelines || AzurePipelines.IsRunningOnAzurePipelinesHosted)
		{
			Information("Publish test result.....................................");
			if (testResultsFiles.Any())
			{
				AzurePipelines.Commands.PublishTestResults(
					new AzurePipelinesPublishTestResultsData
					{
						TestResultsFiles = testResultsFiles.ToArray(),
						TestRunner = AzurePipelinesTestRunnerType.NUnit
					}
				);
			}
			if (vsTestResultsFiles.Any())
			{
				AzurePipelines.Commands.PublishTestResults(
					new AzurePipelinesPublishTestResultsData
					{
						TestResultsFiles = vsTestResultsFiles.ToArray(),
						TestRunner = AzurePipelinesTestRunnerType.VSTest
					}
				);
			}
		}

but As you can see above code, I write two publish method that is very bad, So I want to find a way that can send all test result in one command

thanks

**Comment by pascalberger** [comment]
`PublishTestResults` is just a wrapper around Azure Pipelines service message. Azure Pipelines needs to know what to do with the files and therefore you pass the format. There's no service message which can pass multiple files with different formats. 

> I write two publish method that is very bad

What's the issue with this?
